collector/setting.go

```diff
@@ -0,0 +1,31 @@
+package collector
+
+import (
+	rancher "github.com/rancher/types/client/management/v3"
+	log "github.com/sirupsen/logrus"
+)
+
+func SetSetting(client *rancher.Client, key string, value string) error {
+	setting, err := client.Setting.ByID(key)
+	if err == nil {
+		_, err = client.Setting.Update(setting, map[string]interface{}{"value": value})
+		if err == nil {
+			log.Debugf("UpdateSetting(%s,%s)", key, value)
+		} else {
+			log.Debugf("UpdateSetting(%s,%s): Error: %s", key, value, err)
+		}
+		return err
+	}
+
+	setting, err = client.Setting.Create(&rancher.Setting{
```

Comment:
Shouldn't this be in else block ? Don't see why you would update and then create.

Reply:
Fixed.

Reply:
Should this all be on line 14 ?

Reply:
You are right. More clear now??